test/ipld-all.js

```diff
@@ -52,57 +55,45 @@ describe('IPLD Resolver for dag-cbor + dag-pb', () => {
         cidCbor = cid
 
         each([
-          { node: nodePb, cid: cidPb },
-          { node: nodeCbor, cid: cidCbor }
+          { node: nodePb, format: multicodec.DAG_PB, cidVersion: 0 },
+          { node: nodeCbor, format: multicodec.DAG_CBOR, cidVersion: 1 }
         ], (nac, cb) => {
-          resolver.put(nac.node, { cid: nac.cid }, cb)
+          resolver.put(nac.node, nac.format, {
+            cidVersion: nac.cidVersion
+          }).then(
+            () => cb(null),
+            (error) => cb(error)
+          )
         }, cb)
       }
     ], done)
   })
 
-  it('resolve through different formats', (done) => {
-    resolver.get(cidCbor, 'pb/Data', (err, result) => {
-      expect(err).to.not.exist()
-      expect(result.value).to.eql(Buffer.from('I am inside a Protobuf'))
-      done()
-    })
-  })
+  it('resolve through different formats', async () => {
+    const result = resolver.resolve(cidCbor, 'pb/Data')
+    const [node1, node2] = await result.all()
 
-  it('does not store nodes when onlyHash is passed', (done) => {
-    waterfall([
-      (cb) => dagPB.DAGNode.create(Buffer.from('Some data here'), cb),
-      (node, cb) => resolver.put(node, {
-        onlyHash: true,
-        version: 1,
-        hashAlg: 'sha2-256',
-        format: 'dag-pb'
-      }, cb),
-      (cid, cb) => resolver.bs._repo.blocks.has(cid, cb)
-    ], (error, result) => {
-      if (error) {
-        return done(error)
-      }
+    expect(node1.remainderPath).to.eql('Data')
+    expect(node1.value).to.eql(cidPb)
 
-      expect(result).to.be.false()
-      done()
-    })
+    expect(node2.remainderPath).to.eql('')
+    expect(node2.value).to.eql(Buffer.from('I am inside a Protobuf'))
   })
 
-  it('does not store nodes when onlyHash is passed and a CID is passed', (done) => {
-    const cid = new CID('QmTmxQfEHbQzntsXPTU4ae2ZgBGwseBmS12AkZnKCkuf2G')
-
+  it('does not store nodes when onlyHash is passed', (done) => {
     waterfall([
       (cb) => dagPB.DAGNode.create(Buffer.from('Some data here'), cb),
-      (node, cb) => resolver.put(node, {
-        onlyHash: true,
-        cid
-      }, cb),
-      (cid2, cb) => {
-        expect(cid2).to.equal(cid)
-
-        resolver.bs._repo.blocks.has(cid2, cb)
-      }
+      (node, cb) => {
```

Comment:
this is looking awkward within a `waterfall`, maybe it should be converted to an async test?

Reply:
The tests were just adapted with minimal changes. The tests are in a bad shape currently and need a bigger refactor (there's lots of duplicated code). But I'd like to do it after this change is merged.